magnet.go

```diff
@@ -70,190 +86,220 @@ func (m *Magnet) printHeader() {
 	if m.Version != "" {
 		fmt.Println("Version: ", m.Version)
 	}
-
-	if m.BuildDir != "" {
-		fmt.Println("Build:   ", m.BuildDir)
-	}
-	if m.BuildDir != "" {
-		fmt.Println("Cache:   ", m.CacheDir())
+	if m.CacheDir != "" {
+		fmt.Println("Cache:   ", m.cacheDir())
 	}
 }
 
+// Magnet describes the root logger
 type Magnet struct {
 	Config
 
-	Vertex *progressui.Vertex
-	parent *Magnet
-	status chan *progressui.SolveStatus
-
+	status       chan *progressui.SolveStatus
 	statusLogger *SolveStatusLogger
+	root         MagnetTarget
 
-	cached bool
+	wg  sync.WaitGroup
+	ctx context.Context
+	// cancel cancels the logger process
+	cancel         context.CancelFunc
+	initOutputOnce sync.Once
 }
 
-var root *Magnet
-var rootOnce sync.Once
-var outputOnce sync.Once
+// MagnetTarget describes a child logging target
+//nolint:revive // TODO(dima): rename to Target
+type MagnetTarget struct {
+	root   *Magnet
+	vertex *progressui.Vertex
+	cached bool
+}
 
 // Root creates a root vertex for executing and capturing status of each build target.
-func Root(c Config) *Magnet {
-	const statusChanSize = 128
+func Root(c Config) (*Magnet, error) {
+	if root != nil {
```

Comment:
Instead of `sync.Once` and idempotency (concurrency is less of an issue, I think) the expected behavior would be to fail on subsequent initialization since only a single root target is expected at any time.